create codeql-analysis.yml file

joeniquette opened the pull request:
Testing codeql scanning

## Description
Enable codeql to scan all JS within the repo

## Testing done
N/A

## Screenshots


## Acceptance criteria
- [ ] conducts code scans during every master branch commit
- [ ] Does not block merging into master
- [ ]  provides value add to detecting vulnerabilities of source code

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

joeniquette committed:
create codeql-analysis.yml file

Testing codeql scanning

f1337 reviewed (approved):
Excellent! :metal:
![Bill & Ted playing air guitar, saying 'Excellent!'](https://media.giphy.com/media/l46CDHTqbmnGZyxKo/giphy.gif)

joeniquette commented:
Results have already been posted for the initial [scan](https://github.com/department-of-veterans-affairs/vets-website/security/code-scanning?query=tool%3ACodeQL+is%3Aopen+ref%3Arefs%2Fpull%2F14815%2Fmerge).

billfienberg commented on .github/workflows/codeql-analysis.yml:
```diff
@@ -0,0 +1,68 @@
+# For most projects, this workflow file will not need changing; you simply need
```
>For example, by default, the workflow file for CodeQL code scanning is called `codeql-analysis.yml`. - https://docs.github.com/en/free-pro-team@latest/github/finding-security-vulnerabilities-and-errors-in-your-code/configuring-code-scanning#editing-a-code-scanning-workflow

billfienberg commented:
Relevant Links:
- https://docs.github.com/en/free-pro-team@latest/github/finding-security-vulnerabilities-and-errors-in-your-code/configuring-code-scanning#editing-a-code-scanning-workflow
- https://github.com/github/codeql-action

billfienberg commented on .github/workflows/codeql-analysis.yml:
```diff
@@ -0,0 +1,68 @@
+    # The branches below must be a subset of the branches above
+    branches: [ master ]
+  schedule:
+    - cron: '17 23 * * 1'
```
This means it would run “At 23:17 on Monday," correct? If so, it would run approximately on Monday evening in North American time, correct? 

![image](https://user-images.githubusercontent.com/6130520/98019353-40ee7980-1dc7-11eb-9624-6f779dabfae6.png)

https://crontab.guru/#17_23_*_*_1

joeniquette replied:
yes thats correct. We can change this to anything we want.

mchelen-gov commented on .github/workflows/codeql-analysis.yml:
```diff
@@ -0,0 +1,68 @@
+
+on:
+  push:
+    branches: [ master, brand-consolidation, production ]
```
might want to clean up these branch names since they don't exist

joeniquette replied:
I meant to follow up on this as well. Is there a branch other than master we should be regularly scanning? If not I'll delete the other two and use master.

mchelen-gov replied:
i think `master` is the only long lived branch, the rest are feature branches. idk if we want to scan those during dev work but probably master only is a good starting place

mchelen-gov commented on .github/workflows/codeql-analysis.yml:
```diff
@@ -0,0 +1,68 @@
+
+    # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
+    # If this step fails, then you should remove it and run the build manually (see below)
+    - name: Autobuild
```
is the autobuild successful, or do we need `yarn install` `yarn build` steps?

joeniquette replied:
Since this is just for JS we dont have any build requirements. Autobuild will be fine for this repo.

mchelen-gov replied:
ok cool, wasn't sure if it picked up the `.nvmrc` and used `yarn.lock` instead of NPM as i've had problems with these things in the past

joeniquette committed:
removed all by master for on push branches

If we want to add more branches for on push we can, for now leaving it just at master.

joeniquette committed:
removed all by master for on push branches

If we want to add more branches for on push we can, for now leaving it just at master.

mchelen-gov commented:
@joeniquette can you document the [config location](https://github.com/department-of-veterans-affairs/vets-website/blob/master/.github/workflows/codeql-analysis.yml) and the [results url](https://github.com/department-of-veterans-affairs/vets-website/security/code-scanning?query=tool%3ACodeQL) someplace?

joeniquette commented:
@mchelen-gov I'm guessing you mean some type of documentation within the repo correct? Any particular location which is best suited for this that I can put this in?

mchelen-gov commented:
@joeniquette yeah either some background & links in the codeql-analysis.yml file itself or in https://github.com/department-of-veterans-affairs/vets-website/tree/master/docs

joeniquette commented:
Good idea @mchelen-gov I've created [a ticket](https://app.zenhub.com/workspaces/vsp-identity-5f5bab705a94c9001ba33734/issues/department-of-veterans-affairs/va.gov-team/16280) for myself to pick this up next week.
